cli/azd/internal/cmd/add/add.go

```diff
@@ -166,6 +167,35 @@ func (a *AddAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 		return nil, fmt.Errorf("setting resource: %w", err)
 	}
 
+	// Dependent resources (both existing and to be added)
+	dependentResources := project.GetRequiredDependencies(resourceToAdd)
+	// New dependent resources to be added
+	dependentResourcesToAdd := make([]*project.ResourceConfig, 0)
```

Comment:
I suspect the code actually looks "simpler" if we just treat `resourcesToAdd` as an array that gets filled with dependencies.

And to track the dependency relationship (used various places for printing), we can use a `map[*project.ResourceConfig]*project.ResourceConfig`

Reply:
Good point! Yes it did simplify things quite a bit, and I was able to forgo the dependency map